### PR TITLE
[deconz] Adjust thread name for web socket client

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
A counter is maybe not the best option because a new thread pool will be created each time the thing handler is initialized.

**Comment by lolodomo** [comment]
Whatever the provided name, a new thread pool is created by OH core framework.
Having thread pools with same name is probably allowed.

In case thing UID is too big, we could first try with the thing ID.

PS: I am asking myself if the created thread pool is released when the web socket client is abandoned.

File: bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java [hunk 1]
```diff
@@ -87,10 +87,11 @@ public DeconzBridgeHandler(Bridge thing, WebSocketFactory webSocketFactory, Asyn
...
         String websocketID = thing.getUID().getAsString().replace(':', '-');
-        if (websocketID.length() < 4) {
-            websocketID = "openHAB-deconz-" + websocketID;
-        } else if (websocketID.length() > 20) {
-            websocketID = websocketID.substring(websocketID.length() - 20);
+        if (websocketID.length() < 4 || websocketID.length() > 20) {
+            websocketID = "deconz-" + thing.getUID().getId();
+        }
+        if (websocketID.length() > 20) {
+            websocketID = "deconz";
```
Comment (J-N-K) [inline bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java:94]: That's not a good idea. It should be unique, but repeatable. Maybe hash the full ID and take the last 6 bytes as hex (12 characters + `deconz-` prefix makes a total of 19 characters). I doubt that we'll run into collisions with that and it's nicely reproducible.

Comment (lolodomo) [reply bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java:94]: It is better if it is unique but not required.
But your idea consisting in hashing the UID is good.

Comment (lolodomo) [reply bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java]: Done

Comment (lolodomo) [reply bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java]: Wait a second...

Comment (lolodomo) [reply bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java]: That is ok now

**Comment by lolodomo** [comment]
Awaiting a new method in core framework.

**Comment by lolodomo** [comment]
Awaiting PR openhab/openhab-core#3359 to be merged to then update this PR

**Comment by lolodomo** [comment]
My initial analysis was wrong, there is no risk of IAE with the current code.

**Review by wborn — APPROVED** [review]

**Pull request merged**